### PR TITLE
vendor: upgrade k8s deps

### PR DESCRIPTION
**Pull request by nicks**
Hello @milas,

Please review the following commits I made in branch nicks/vendor:

e1d6e62095958cbee5a1a43df139ffb1362776d7 (2022-01-18 13:02:06 -0500)
vendor: upgrade k8s deps

Code review reminders, by giving a LGTM you attest that:

* Commits are adequately tested
* Code is easy to understand and conforms to style guides
* Incomplete code is marked with TODOs
* Code is suitably instrumented with logging and metrics

### CODE REVIEW

**Comment by nicks** [comment]
uggggh looks like CI is hitting https://github.com/kubernetes/kubernetes/pull/107452

fortunately, it's getting pulled into the 1.23 release branch https://github.com/kubernetes/apiserver/tree/release-1.23, so maybe we can cherry-pick it without too much trouble...

**Comment by nicks** [comment]
ok, added a cherrypick

**Comment by nicksieger** [comment]
When we do these upgrades does it affect which k8s versions Tilt will work with? I see that the [version skew policy](https://kubernetes.io/releases/version-skew-policy/) states that kubectl should be compatible within one minor version of the apiserver. In practice I think kubectl has been pretty stable; I recall running with a newer kubectl on clusters that were definitely 2 or more minor versions old. Wondering if we should have some sort of a guideline or statement for k8s compatibility per Tilt version.

**Comment by nicks** [comment]
ya, the client-go compatibility matrix is much broader - https://github.com/kubernetes/client-go#compatibility-matrix

my general understanding is that the "guaranteed" version skew is very narrow, but that the practical version skew is very broad.

File: internal/controllers/builder.go [hunk 1]
```diff
@@ -41,6 +60,7 @@ func (c *ControllerBuilder) OnChange(_ context.Context, _ store.RStore, _ store.
 func (c *ControllerBuilder) SetUp(ctx context.Context, st store.RStore) error {
 	mgr := c.tscm.GetManager()
 	client := c.tscm.GetClient()
+	_ = logger.Get(ctx)
```
Comment (milas) [inline internal/controllers/builder.go:63]: Is this being done for side-effects (panic if no logger attached)? If so, a comment might be warranted

**Review by milas — APPROVED** [review]

File: go.mod [hunk 2]
```diff
@@ -223,22 +222,18 @@ require (
...
+	sigs.k8s.io/structured-merge-diff/v4 v4.2.0 // indirect
 )
 
 replace (
 	github.com/Azure/go-autorest => github.com/Azure/go-autorest v14.2.0+incompatible
 
-	// This is just evanphx/json-patch v4.6.0 with a fix for
-	// https://github.com/evanphx/json-patch/issues/98
-	// so that we can pull it in correctly
-	github.com/evanphx/json-patch => github.com/tilt-dev/json-patch/v4 v4.8.1 // indirect
```
[the remark this answers is not in the document]
Comment (nicks) [reply go.mod:240]: done!

**Pull request merged**